lisp/tree-sitter.el

```diff
@@ -272,14 +287,21 @@ Both SETUP-FUNCTION and TEARDOWN-FUNCTION should be idempotent."
 (define-obsolete-function-alias 'tree-sitter-node-at-point 'tree-sitter-node-at-pos "2021-08-30")
 
 ;;;###autoload
-(defun tree-sitter-node-at-pos (&optional node-type pos)
+(defun tree-sitter-node-at-pos (&optional node-type pos strict)
   "Return the smallest syntax node of type NODE-TYPE at POS.
 NODE-TYPE may be a symbol, corresponding to a named syntax node; a string,
 corresponding to an anonymous node, or a keyword, holding a special value. For
 the special value `:named', return the smallest named node at POS. For the
 special value `:anonymous', return the smallest anonymous node at POS. IF POS is
-nil, POS defaults to the point. Whenever NODE-TYPE is non-nil (other than
-`:named'), it is possible for the function to return nil."
+nil, POS defaults to the point. When STRICT is non-nil, signal an error when a
+specified named NODE-TYPE does not exist in the current grammar. Whenever
+NODE-TYPE is non-nil (other than `:named'), it is possible for the function
+to return nil."
+  (when (and strict node-type
+         (symbolp node-type)
+         (not (keywordp node-type)))
+    (unless (memq node-type tree-sitter--named-node-types)
+      (error "No such named node-type: %s" node-type)))
```

Comment:
We should define a new error symbol, so that it can be handled/ignored specifically. The list of current error symbols is [here](https://github.com/emacs-tree-sitter/elisp-tree-sitter/blob/0.15.1/core/src/error.rs). (These are defined in Rust because they are signaled there, but this one should be in Lisp.)

Reply:
Anonymous node types should also be checked. `tsc-lang-node-type-id` also works on them.